my-webxr-app/src/components/DataPoint.tsx

```diff
@@ -38,26 +42,35 @@ export default function DataPoint({
         if (selectedDataPoint?.id === id) {
           setSelectedDataPoint(null);
 
-        // Update point to be selected and set its fields
+          // Update point to be selected and set its fields
         } else {
           setSelectedDataPoint({
-            id, marker, color, columnX, columnY, columnZ, meshProps,
+            id, marker, color, columnX, columnY, columnZ, positions,
           });
         }
       }}
     >
-      {/* This first mesh stores custom data about the DataPoint */}
-      <mesh
-        userData={{
-          id, columnX, columnY, columnZ, marker, color,
-        }}
-        {...meshProps}
-      >
-        {/* Low numbers to try to minimize the number of faces we need to render */}
-        {/* There will be a LOT of these present in the simulation */}
-        <sphereGeometry args={size} />
-        <meshStandardMaterial />
-      </mesh>
+
+      <points>
+        <bufferGeometry attach="geometry">
+          <bufferAttribute
+            attach="attributes-position"
+            array={positions}
+            count={positions.length / 3} //
+            itemSize={3}
+          />
+        </bufferGeometry>
+        <pointsMaterial
+          attach="material"
+          map={CircleImg}
+          color={0x000000}
+          size={0.01}
+          sizeAttenuation
+          transparent={false}
+          alphaTest={0.5}
+          opacity={1.0}
+        />
+      </points>
```

Comment:
The `Points` class is a great idea. Are they still interactive like this? We'll have to see what others optimizations may be possible.

Reply:
Not immediately. There would be more work to do than we have time to accomplish.